lang/cs/Org.Apache.REEF.Wake/Remote/Impl/WritableLink.cs

```diff
@@ -174,15 +174,16 @@ public T Read()
                 return default(T);
             }
 
-            T value = _cache.GetInstance(dataType);
-
-            if (value == null)
+            try
+            {
+                T value = (T) _injector.ForkInjector().GetInstance(dataType);
```

Comment:
You always want to get new instantiation. Thats why.

On Wed, May 13, 2015 at 11:41 AM, Julia Wang <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.Wake/Remote/Impl/WritableLink.cs
> <https://github.com/apache/incubator-reef/pull/181#discussion_r30262120>:
>
> > @@ -174,15 +174,16 @@ public T Read()
> >                  return default(T);
> >              }
> >
> > -            T value = _cache.GetInstance(dataType);
> > -
> > -            if (value == null)
> > +            try
> > +            {
> > +                T value = (T) _injector.ForkInjector().GetInstance(dataType);
>
> Why fork here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/181/files#r30262120>.
>